maco/cli.py

```diff
@@ -22,6 +23,19 @@ def process_file(
     force: bool,
 ):
     """Process a filestream with the extractors and rules."""
+    unneutered = io.BytesIO()
+    try:
+        cart.unpack_stream(stream, unneutered)
+    except Exception:
+        # use original stream if anything goes wrong here
+        # i.e. invalid/malformed cart
+        pass
+    else:
+        # use unneutered stream
+        stream = unneutered
```

Comment:
Think `cart.is_cart()` should cover this kind of check.

If the file is a 'valid' CaRT file, then unpack otherwise use original stream to proceed with analysis.

Reply:
Thanks, I want to avoid the `is_cart()` function since it requires a byte string rather than a stream, which means that I need to read the whole file into memory. I don't really want to rely on internal knowledge of the CART library by supplying the first 38 bytes to this function, just in case something changes in the future.

Since cart raises the `InvalidCARTException` on an error with the cart format, I figure it is simpler to default to original stream analysis on any exception. Including the theoretical case of when a file appears to be a standard cart with the header, but turns out to be something else.